core/cas-server-core-authentication/src/main/java/org/apereo/cas/authentication/handler/support/AbstractUsernamePasswordAuthenticationHandler.java

```diff
@@ -58,7 +58,6 @@ protected HandlerResult doAuthentication(final Credential credential) throws Gen
         }
 
         userPass.setUsername(transformedUsername);
-        userPass.setPassword(this.passwordEncoder.encode(userPass.getPassword()));
```

Comment:
Gotcha. In that case, as this breaks functionality, you're likely to refactor the APIs to come up with enough abstractions/components to handle this behavior and allow for proper encoding across all, and then make exceptions for those that cannot.